packages/@aws-cdk/aws-ecs/lib/queue-worker-service-base.ts

```diff
@@ -0,0 +1,180 @@
+   *
+   * @default none
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * A name for the queue.
+   *
+   * If specified and this is a FIFO queue, must end in the string '.fifo'.
+   *
+   * @default 'ecs-worker-service-queue'
+   */
+  readonly queueName?: string;
+
+  /**
+   * Maximum capacity to scale to.
+   *
+   * @default (desiredTaskCount * 2)
+   */
+  readonly maxScalingCapacity?: number
+
+  /**
+   * The intervals for scaling based on the SQS queue's ApproximateNumberOfMessagesVisible metric.
+   *
+   * Maps a range of metric values to a particular scaling behavior.
+   * https://docs.aws.amazon.com/autoscaling/ec2/userguide/as-scaling-simple-step.html
+   *
+   * @default [{ upper: 0, change: -1 },{ lower: 100, change: +1 },{ lower: 500, change: +5 }]
+   */
+  readonly scalingSteps?: autoscaling.ScalingInterval[];
+}
+
+/**
+ * Base class for a Fargate and ECS queue worker service
+ */
+export abstract class QueueWorkerServiceBase extends cdk.Construct {
+  /**
+   * The sqs queue that the worker service will process from
+   */
+  public readonly sqsQueue: Queue;
+
+  // Properties that have defaults defined. The Queue Worker will handle assigning undefined properties with default
+  // values so that derived classes do not need to maintain the same logic.
+
+  /**
+   * Environment variables that will include the queue name
+   */
+  public readonly environment: { [key: string]: string };
+  /**
+   * The minimum number of tasks to run
+   */
+  public readonly desiredCount: number;
+  /**
+   * The maximum number of instances for autoscaling to scale up to
+   */
+  public readonly maxCapacity: number;
+  /**
+   * The scaling interval for autoscaling based off an SQS Queue size
+   */
+  public readonly scalingSteps: autoscaling.ScalingInterval[];
+  /**
+   * The AwsLogDriverto use for logging if logging is enabled.
+   */
+  public readonly logDriver?: LogDriver;
+
+  constructor(scope: cdk.Construct, id: string, props: QueueWorkerServiceBaseProps) {
+    super(scope, id);
+
+    // Create the worker sqs queue
+    const queueName = props.queueName !== undefined ? props.queueName : 'ecs-worker-service-queue';
+    this.sqsQueue = new sqs.Queue(this, 'ecs-worker-service-queue', {
+      queueName
+    });
+
+    // Setup AutoScaling scaling intervals
+    const defaultScalingSteps = [{ upper: 0, change: -1 }, { lower: 100, change: +1 }, { lower: 500, change: +5 }];
+    this.scalingSteps = props.scalingSteps !== undefined ? props.scalingSteps : defaultScalingSteps;
+
+    // Create log driver if logging is enabled
+    const enableLogging = props.enableLogging !== undefined ? props.enableLogging : true;
+    this.logDriver = enableLogging ? this.createAwsLogDriver(this.node.id) : undefined;
+
+    // Add the queuename to environment variables
+    this.environment = props.environment !== undefined ?
+      this.addQueueNameToEnvironment(props.environment) : {QUEUE_NAME: this.sqsQueue.queueName};
```

Comment:
Style nit: spaces around curly braces, e.g. `{ QUEUE_NAME: this.sqsQueue.queueName }`

Reply:
updated